python/cudf/cudf/core/index.py

```diff
@@ -1411,24 +1466,186 @@ def __cuda_array_interface__(self):
         return self._values.__cuda_array_interface__
 
 
+class BaseNumericIndex(GenericIndex):
```

Comment:
Maybe `BaseNumericIndex` wasn't the best name - sorry for suggesting it. Just `NumericIndex` is fine.